lib/koans/09_pattern_matching.ex

```diff
@@ -0,0 +1,62 @@
+defmodule PatternMatching do
+  use Koans
+
+  koan "one matches one" do
+    assert match?(1, 1)
+  end
+
+  koan "patterns can be used to pull things apart" do
+    [head | _tail] = [1,2,3,4]
+
+    assert head == 1
```

Comment:
I'd keep `tail` and also assert on it. 
Then introduce `_something` as away to explicitly ignore things.